test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java

```diff
@@ -504,16 +506,18 @@ public void afterEach(ExtensionContext context) throws Exception {
         if (!failedBoot) {
             popMockContext();
             ClassLoader original = setCCL(runningQuarkusApplication.getClassLoader());
-            for (Object afterEachCallback : afterEachCallbacks) {
-                Map.Entry<Class<?>, ?> tuple = createQuarkusTestMethodContextTuple(context);
-                afterEachCallback.getClass().getMethod("afterEach", tuple.getKey())
-                        .invoke(afterEachCallback, tuple.getValue());
-            }
             try {
+                for (Object afterEachCallback : afterEachCallbacks) {
+                    Map.Entry<Class<?>, ?> tuple = createQuarkusTestMethodContextTuple(context);
+                    afterEachCallback.getClass().getMethod("afterEach", tuple.getKey())
+                            .invoke(afterEachCallback, tuple.getValue());
+                }
```

Comment:
Running this outside of the TCCL restoration block looked wrong, so I moved it inside.

Reply:
Not an actual one, but compare it with this:
https://github.com/quarkusio/quarkus/blob/2.7.2.Final/test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java#L407

Why should the TCCL only be restored if there is an exception in a beforeEachCallback, but _not_ if there is an exception in an afterEachCallback? Doesn't make sense to me.

Reply:
PS: afterAll also restores the TCCL just like beforeEach. Same happens for beforeClassCallback. 🤷‍♂️

Reply:
This change looks right to me